tests/system/python/api/test_authentication.py

```diff
@@ -0,0 +1,212 @@
+                             headers={"authorization": TOKEN})
+                r = conn.getresponse()
+                assert 200 == r.status
+                r = r.read().decode()
+                jdoc = json.loads(r)
+                assert {'message': 'User with id:<3> has been updated successfully'} == jdoc
+
+        def test_delete_user(self, foglamp_url):
+                conn = http.client.HTTPConnection(foglamp_url)
+                conn.request("DELETE", "/foglamp/admin/4/delete", headers={"authorization": TOKEN})
+                r = conn.getresponse()
+                assert 200 == r.status
+                r = r.read().decode()
+                jdoc = json.loads(r)
+                assert {'message': "User has been deleted successfully"} == jdoc
+
+        def test_logout_all(self, foglamp_url):
+                conn = http.client.HTTPConnection(foglamp_url)
+                conn.request("PUT", '/foglamp/1/logout', headers={"authorization": TOKEN})
+                r = conn.getresponse()
+                assert 200 == r.status
+                r = r.read().decode()
+                jdoc = json.loads(r)
+                assert jdoc['logout']
+
+        def test_admin_actions_reg_user(self, foglamp_url):
+                """Test that regular user is not able to perform any actions that only an admin can"""
+                # Login with regular user
+                conn = http.client.HTTPConnection(foglamp_url)
+                conn.request("POST", "/foglamp/login", json.dumps({"username": "user", "password": "foglamp"}))
+                r = conn.getresponse()
+                assert 200 == r.status
+                r = r.read().decode()
+                jdoc = json.loads(r)
+                assert not jdoc['admin']
+                _token = jdoc["token"]
+
+                # Create User
+                conn.request("POST", "/foglamp/admin/user", body=json.dumps({"username": "other", "password": "User@123"}),
+                             headers={"authorization": _token})
+                r = conn.getresponse()
+                assert 403 == r.status
+                r = r.read().decode()
+                assert "403: Forbidden" == r
+
+                # Reset User
+                conn.request("PUT", "/foglamp/admin/2/reset", body=json.dumps({"role_id": 1, "password": "F0gl@p!"}),
+                             headers={"authorization": _token})
+                r = conn.getresponse()
+                assert 403 == r.status
+                r = r.read().decode()
+                assert "403: Forbidden" == r
+
+                # Delete User
+                conn.request("DELETE", "/foglamp/admin/2/delete", headers={"authorization": _token})
+                r = conn.getresponse()
+                assert 403 == r.status
+                r = r.read().decode()
+                assert "403: Forbidden" == r
+
+        def test_login_with_certificate(self, foglamp_url):
```

Comment:
It would be great to have a test_admin_login_with_certificate, and rename this to test_user_login_with_certificate.

And create some cert at runtime and use that to test valid cert login